libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5169,6 +5169,30 @@ MAV_RESULT GCS_MAVLINK::handle_do_set_safety_switch_state(const mavlink_command_
     }
 }
 
+MAV_RESULT GCS_MAVLINK::handle_do_set_sys_cmp_id(const mavlink_command_int_t &packet, const mavlink_message_t &msg)
+{
+    const uint8_t new_system_id = packet.param1;
+    const uint8_t new_component_id = packet.param2;
+    const uint8_t do_reboot = packet.param3;
```

Comment:
Should this be a "reboot" or "this change should take effect immediately" flag?

Reply:
This is a reboot flag.

Reply:
Yep, we read the message description.

The *intent* is for the thing to take effect immediately, 'though, isn't it?  We can *probably* do that without a reboot my modifying the mavlink state variables.

I think this field should probably be "take effect immediately", rather than "reboot".  We can discuss at DevCallEU tonight if you like....